### ISSUE TITLE
YouTube Links Not Working

### ISSUE THREAD

**Issue by RexfelisLOC** [MyToolkit/MyToolkit#103]
It seems an update is needed for YouTube video link changes.

**Comment by RexfelisLOC** [comment]
VideoId example "o_1aF54DO60"

Seems to be working for some videos and not others.

**Comment by cansado2930** [comment]
I have the same problem, have you found What is the problem?

**Comment by Deery5000** [comment]
Still no fix to this yet?
All links are down regardless of the quality :(

**Comment by BRC2016** [comment]
A month ago my YouTube player was working great now I get Error: unsupported video type or invalid path.  
                   YouTubeId="xxxxxxxxxx",
                   Title=await YouTube.GetVideoTitleAsync("xxxxxxxxxx"),
                   ThumbnailUri= YouTube.GetThumbnailUri("xxxxxxxxxx",YouTubeThumbnailSize.Medium),
                   VideoUri=await YouTube.GetVideoUriAsync("xxxxxxxxxx",YouTubeQuality.Quality360P)

GetVideoTitleAsync works still, as does GetThumbnailUri but the GetVideoUriAsync yields gibberish. Is this the same issue as others are facing?

**Comment by Deery5000** [comment]
@BRC2016 
yes we're experiencing the same issue, painful for my app Cinema Box. I do enjoy trailers 

Not seen a response here in a while, I guess their not planning on updating this :(

**Comment by RicoSuter** [comment]
I'm not planning into looking into this error as it always breaks and I don't use the feature anymore... but I'll accept PRs...

**Comment by BRC2016** [comment]
@RicoSuter 
I respect that decision.  Can you suggest an alternative method to embed YouTube videos in a UWP app.

**Comment by Deery5000** [comment]
There is a ton of people who use this API, can you not fix this?

**Comment by BRC2016** [comment]
@Deery5000 I found that the best solution is to use the embedable Youtube player <iframe> in an HTML page and then call via web view.  https://developers.google.com/youtube/player_parameters.

**Comment by Deery5000** [comment]
Thanks for getting back to me. I found an alternative which is performing as expected. Thank you On 1 Jul 2019 00:12, BRC2016 <notifications@github.com> wrote:@Deery5000 I found that the best solution is to use the embedable Youtube player <iframe> in an HTML page and then call via web view.  https://developers.google.com/youtube/player_parameters.

**Comment by cansado2930** [comment]
the problem is that it can not get a valid url in order to get the JavaScript code to generate the signature.

**Comment by tejaat1995** [comment]
Hello Everyone,

I am getting this exception while using **var url = await YouTube.GetVideoUriAsync(youTubeId, YouTubeQuality.Quality720P);**

**MyToolkit.Multimedia.YouTubeUriNotFoundException: No matching YouTube video or audio stream URI could be found. The video may not be available in your country, is private or uses RTMPE (protected streaming).**

Can anyone help me with this?
